packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,156 @@
+  implements ParametersGatherer<ComponentName> {
+  private sourceFsPath: string;
+  constructor(sourceFsPath: string) {
+    this.sourceFsPath = sourceFsPath;
+  }
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ComponentName>
+  > {
+    const inputOptions = {
+      value: getComponentName(getComponentPath(this.sourceFsPath)),
+      placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+      promopt: nls.localize(RENAME_INPUT_PROMPT)
+    } as vscode.InputBoxOptions;
+    const inputResult = await vscode.window.showInputBox(inputOptions);
+    return inputResult
+      ? { type: 'CONTINUE', data: { name: inputResult } }
+      : { type: 'CANCEL' };
+  }
+}
+
+async function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = getComponentName(componentPath);
+  await checkForDuplicateName(componentPath, newName);
+  const items = await fs.promises.readdir(componentPath);
+  for (const item of items) {
+    // only rename the file that has same name with component
+    if (isNameMatch(item, componentName, componentPath)) {
+      const newItem = item.replace(componentName, newName);
+      await fs.promises.rename(
+        path.join(componentPath, item),
+        path.join(componentPath, newItem)
+      );
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  await fs.promises.rename(
+    componentPath,
+    newComponentPath
+  );
+  notificationService.showWarningMessage(nls.localize(RENAME_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function getComponentName(componentPath: string): string {
+  return path.basename(componentPath);
+}
+
+async function checkForDuplicateName(componentPath: string, newName: string) {
+  const isNameDuplicate = await isDuplicate(componentPath, newName);
+  if (isNameDuplicate) {
+    const errorMessage = nls.localize(RENAME_INPUT_DUP_ERROR);
+    notificationService.showErrorMessage(errorMessage);
+    throw new Error(format(errorMessage));
+  }
+}
+
+async function isDuplicate(componentPath: string, newName: string): Promise<boolean> {
+  // A LWC component can't share the same name as a Aura component
+  const componentPathDirName = path.dirname(componentPath);
+  let lwcPath: string;
+  let auraPath: string;
+  if (isLwcComponent(componentPath)) {
+    lwcPath = componentPathDirName;
+    auraPath = path.join(path.dirname(componentPathDirName), AURA);
+  } else {
+    lwcPath = path.join(path.dirname(componentPathDirName), LWC);
+    auraPath = componentPathDirName;
+  }
+  const allLwcComponents = await fs.promises.readdir(lwcPath);
+  const allAuraComponents = await fs.promises.readdir(auraPath);
+  return allLwcComponents.includes(newName) || allAuraComponents.includes(newName) ? true : false;
```

Comment:
nit: here you can just do 
`return (allLwcComponents.includes(newName) || allAuraComponents.includes(newName));